### PR TITLE
improve search fields and some buttons alignement

### PR DESCRIPTION
**Pull request by Nilvus**
I didn't have time to review #9712 as requested by @elstoc. His proposal is good and was needed. Anyway, I've also seen that new search fields in shortcut view by @dterrahe needed some CSS tweaks. I so see that same padding was needed for those search box, so better use .search to apply to all search fields. I also find a way to have a better (I hope) balanced padding around search image inside search entry. See:

before:
 

after:
 

I also improve some buttons alignment regardless main view above them in shortcuts and presets view in preferences window.

See:

before:

after:

### CODE REVIEW

**Comment by dterrahe** [comment]
The search boxes should probably also be a bit wider, because they don't automatically rescale with the placeholder text and "search shortcuts list" already gets cut off (and this may be worse in translations).

**Comment by elstoc** [comment]
For the shortcuts section I think I would prefer either:

- Each pane has a header above it and to the left ("actions", "shortcuts"), with the search field to the right. This would reduce vertical real-estate but the window is resizeable so I'm not sure it's a massive problem; or
- Move the search fields and buttons so that they sit in between the two panes. The search box for the upper pane could be 'attached' to it at the bottom-left; The search bottom for the lower pane could be 'attached' to it at the top-right; The buttons would sit in the middle.

As I've mentioned in another PR, I'm still not 100% convinced that the panes are the right way round. For me the actions pane should be at the top, since the selection of an action causes the shortcuts pane to be redrawn.

Obviously all this requires Gtk work as well as CSS.

**Comment by dterrahe** [comment]
I like the second suggestion, so I'll give it a try. The gtkpaned widget only has one handle, so the buttons would be above or below that. Hopefully not too displeasing and hopefully still obvious enough where to drag the divider to resize.

The first option I'm not sure adds much, since the top row of each list already contains the word shortcut or action, so I think it would look repetitive? It would add two rows, since logically the import/export buttons would then need their own rows.

The logic for having shortcuts on top is that that's probably the main list people want to see and somehow I thought it made sense to have the "filter" list below. But I'm happy to swap them if that's the consensus.

**Comment by elstoc** [comment]
> I think it would look repetitive?

I wouldn't necessarily think to look at the first column of the treeview to see what it's for. Even if it's repetitive I still think it would be nice as a navigation aid. However, I do see the issues with screen real-estate. Perhaps if we went for option 2 we could give each of the search boxes an actual label next to the search entry "shortcuts" / "actions", which might be sufficient.

**Comment by Nilvus** [comment]
Except wider answer, other comments are not really related on that PR. I think opening an issue would have been better for a better visibility of those comments that are on UI/UX design and not small adjustments with CSS, purpose of this PR.

I will continue in @dterrahe PR to answer about proposal.

**Comment by elstoc** [comment]
@Nilvus I was just trying to avoid you spending time on something if the layout isn't even finalised yet.

**Comment by TurboGit** [comment]
I see that the entries are not wide enough to contain the text, see "search shortcuts list" (first entry). Can this be fixed in CSS ?

image

**Comment by elstoc** [comment]
It could probably be shortened to search shortcuts and search actions

**Comment by TurboGit** [comment]
> It could probably be shortened to search shortcuts and search actions

No because it becomes ambiguous. You don't search a shortcut but the action name in the shortcut list.

**Comment by Nilvus** [comment]
> @Nilvus I was just trying to avoid you spending time on something if the layout isn't even finalised yet.

Ok



> I see that the entries are not wide enough to contain the text, see "search shortcuts list" (first entry). Can this be fixed in CSS ?

A min-width is possible but it would be good to wait and see what #9726 will become.

**Comment by Nilvus** [comment]
@TurboGit: as #9726 probably needs some time and thoughts to find a good one (I think actual design UI is good and probably #9726 is not needed. Text in search boxes is quite clear and buttons on bottom are a better and more consistent UI), I think this could be merged if min-width just added with last commit is good for you. It's no big deal to work later again with CSS if we find another way that is good for all.

File: data/themes/darktable.css [hunk 1]
```diff
@@ -1272,15 +1272,24 @@ dialog scrollbar
...
-#preferences_box #shortcut_controls
+#preferences_box #shortcut_controls .search
 {
-  margin-left: 0.35em;
+  margin-right: 0.35em;
+}
+
+#preferences_box #shortcut_controls entry
+{
+  min-width: 220px;
```
Comment (TurboGit) [inline data/themes/darktable.css:1282]: wouldn't this be better expressed in em to work ok on HiDPI screen?

**Review by TurboGit — COMMENTED** [review]
Fully agreed for a fix right now, we'll see how the other PR will be handled. Just a little comment for HiDPI screen.

File: data/themes/darktable.css (hunk 1 above)
Comment (Nilvus) [reply data/themes/darktable.css]: I have hesitated but you're right. Better with em. Fixed with last update.

**Review by TurboGit — APPROVED** [review]
All good now, thanks!

**Comment by TurboGit** [comment]
@Nilvus : Just found out that the size of the search entries are ok when the dialog is opened via the preferences but not when the same dialog is opened after clicking on the shortcut icon and clicking on an item in the interface. Can you have a look? TIA.

**Comment by Nilvus** [comment]
> @Nilvus : Just found out that the size of the search entries are ok when the dialog is opened via the preferences but not when the same dialog is opened after clicking on the shortcut icon and clicking on an item in the interface. Can you have a look? TIA.

Shit, forget to check that part by thinking that was the same as preferences one... See #9824 that fix that!

**Pull request merged**